libraries/botbuilder-dialogs/src/dialogHelper.ts

```diff
@@ -95,7 +95,12 @@ export async function runDialog(dialog: Dialog, context: TurnContext, accessor:
  * Helper to determine if we should send an EoC to the parent or not.
  * @param context 
  */
-function sendEoCToParent(context: TurnContext): boolean {
+export function shouldSendEndOfConversationToParent(context: TurnContext, turnResult: DialogTurnResult): boolean {
+    if (!(turnResult.status == DialogTurnStatus.complete || turnResult.status == DialogTurnStatus.cancelled)) {
+        // The dialog is still going, don't return EoC.
+        return false;
+    }
+
```

Comment:
@chon219, this is a combination that combines [`DialogManager.ShouldSendEndOfConversationToParent()`](https://github.com/microsoft/botbuilder-dotnet/blob/2e2f20b65017305f3697d0d5b398dcd2bbfa6a04/libraries/Microsoft.Bot.Builder.Dialogs/DialogManager.cs#L378-L384) and [`DialogExtensions.SendEoCToParent()`](https://github.com/microsoft/botbuilder-dotnet/blob/2e2f20b65017305f3697d0d5b398dcd2bbfa6a04/libraries/Microsoft.Bot.Builder.Dialogs/DialogExtensions.cs#L100), correct?

I think we should also take this change in C#. What do you think @gabog?

Reply:
@stevengum I think so. We have two implementations of this function in c# but it seems that we only need one.